src/Sentry/Internal/Enricher.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Sentry.PlatformAbstractions;
+using Sentry.Reflection;
+using OperatingSystem = Sentry.Protocol.OperatingSystem;
+using Runtime = Sentry.Protocol.Runtime;
+
+namespace Sentry.Internal
+{
+    internal class Enricher
+    {
+        private readonly SentryOptions _options;
+
+        private readonly Lazy<Runtime> _runtimeLazy = new(() =>
+        {
+            var current = PlatformAbstractions.Runtime.Current;
+            return new Runtime
+            {
+                Name = current.Name,
+                Version = current.Version,
+                RawDescription = current.Raw
+            };
+        });
+
+        private readonly Lazy<SdkVersion> _sdkVersionLazy =
+            new(() => typeof(ISentryClient).Assembly.GetNameAndVersion());
+
+        private readonly Lazy<string?> _releaseLazy = new(ReleaseLocator.GetCurrent);
+
+        public Enricher(SentryOptions options)
+        {
+            _options = options;
+        }
+
+        public void Apply(IEventLike eventLike)
+        {
+            // Runtime
+            if (!eventLike.Contexts.ContainsKey(Runtime.Type))
+            {
+                eventLike.Contexts[Runtime.Type] = _runtimeLazy.Value;
+            }
+
+            // Operating System
+            if (!eventLike.Contexts.ContainsKey(OperatingSystem.Type))
+            {
+                // RuntimeInformation.OSDescription is throwing on Mono 5.12
+                if (!PlatformAbstractions.Runtime.Current.IsMono())
+                {
+                    eventLike.Contexts.OperatingSystem.RawDescription = RuntimeInformation.OSDescription;
+                }
+            }
+
+            // SDK
+            // SDK Name/Version might have be already set by an outer package
+            // e.g: ASP.NET Core can set itself as the SDK
+            if (eventLike.Sdk.Version is null && eventLike.Sdk.Name is null)
+            {
+                eventLike.Sdk.Name = Constants.SdkName;
+                eventLike.Sdk.Version = _sdkVersionLazy.Value.Version;
+            }
+
+            if (_sdkVersionLazy.Value.Version is not null)
+            {
+                eventLike.Sdk.AddPackage("nuget:" + _sdkVersionLazy.Value.Name, _sdkVersionLazy.Value.Version);
+            }
+
+            // Release
+            if (eventLike.Release is null)
+            {
+                eventLike.Release = _options.Release ?? _releaseLazy.Value;
+            }
+
+            // Environment
+            if (string.IsNullOrWhiteSpace(eventLike.Environment))
+            {
+                var foundEnvironment = EnvironmentLocator.Locate();
```

Comment:
Can't recall if `EnvironmentLocator.Locate();` caches the result but it should

Reply:
Actually, `Locate()` apparently doesn't. `Current` does. Changed.